crates/spirv-std/src/lib.rs

```diff
@@ -51,6 +51,26 @@ pub use glam;
 pub use num_traits;
 pub use textures::*;
 
+/// Calls the `OpDemoteToHelperInvocationEXT` instruction, which corresponds to discard() in HLSL
+pub fn demote_to_helper_invocation() {
+    #[cfg(target_arch = "spirv")]
+    unsafe {
+        asm!(
+            "OpExtension \"SPV_EXT_demote_to_helper_invocation\"",
+            "OpCapability DemoteToHelperInvocationEXT",
+            "OpDemoteToHelperInvocationEXT"
+        );
+    }
+}
+
+/// Calls the `OpKill` instruction, which corresponds to discard() in GLSL
+pub fn discard() {
```

Comment:
Besides the naming discussion, there's two things I want to add here (tho neither should block the PR):
* would `fn discard() -> !` make sense? i.e. is execution guaranteed to end, *unlike* `demote_to_helper_invocation`?
* long-term we probably want to have some kind of token to avoid being able to call this outside of a fragment shader, right? (if `demote_to_helper_invocation` also needs that, it could be a `demote_to_helper` method on a `ShaderInvocation` token, which I find nice naming-wise, but that's just bikeshedding)